plugins/chat/assets/javascripts/discourse/services/chat-drawer-router.js

```diff
@@ -131,23 +131,74 @@ const ROUTES = {
 export default class ChatDrawerRouter extends Service {
   @service router;
   @service chatHistory;
+  @service chat;
+  @service siteSettings;
+  @service chatStateManager;
+  @service chatChannelsManager;
 
   @tracked component = null;
   @tracked drawerRoute = null;
   @tracked params = null;
 
   routeNames = Object.keys(ROUTES);
 
+  #hasThreads() {
```

Comment:
not sure why most these methods are private

Reply:
I wanted to avoid polluting the public interface mainly because we use the ChatDrawerRouter service. I've changed the getters to be public because that is okay, but the other methods do not make sense to be public.